runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java

```diff
@@ -155,42 +158,54 @@ public void onConfigChanged(ConfigChanged event)
 				frame.setResizable(!config.lockWindowSize());
 			}
 
-			if (!event.getKey().equals("gameSize"))
+			if (client == null)
 			{
 				return;
 			}
 
-			if (client == null)
+			if (event.getKey().equals("gameSize"))
 			{
-				return;
-			}
+				int width = config.gameSize().width;
+				int height = config.gameSize().height;
 
-			int width = config.gameSize().width;
-			int height = config.gameSize().height;
+				// The upper bounds are defined by the applet's max size
+				// The lower bounds are taken care of by ClientPanel's setMinimumSize
 
-			// The upper bounds are defined by the applet's max size
-			// The lower bounds are taken care of by ClientPanel's setMinimumSize
+				if (width > 7680)
+				{
+					width = 7680;
+				}
 
-			if (width > 7680)
-			{
-				width = 7680;
-			}
+				if (height > 2160)
+				{
+					height = 2160;
+				}
 
-			if (height > 2160)
-			{
-				height = 2160;
-			}
+				final Dimension size = new Dimension(width, height);
+				final Dimension clientSize = client.getSize();
 
-			final Dimension size = new Dimension(width, height);
+				// When resizing from the desktop, Java seems to already pack(). Calling it again puts the frame to the
+				// previous size. Thus, (try) to only resize and call pack() when the user sets the size manually from
+				// the sidebar.
+				if (event.getOldValue() == null ||
+						clientSize.height != size.height ||
+						clientSize.width != size.width)
+				{
+					client.setSize(size);
+					client.setPreferredSize(size);
+					client.getParent().setPreferredSize(size);
+					client.getParent().setSize(size);
 
-			client.setSize(size);
-			client.setPreferredSize(size);
-			client.getParent().setPreferredSize(size);
-			client.getParent().setSize(size);
+					if (frame.isVisible())
+					{
+						frame.pack();
+					}
+				}
+			}
 
-			if (frame.isVisible())
+			if (event.getKey().equals("maximized") && config.maximized())
```

Comment:
When will this be reset? You need to set it to previous state too in case you receive such event.